include/seastar/core/coroutine.hh

```diff
@@ -51,10 +53,26 @@ execute_involving_handle_destruction_in_await_suspend(std::invocable<> auto&& fu
 }
 
 
+class coroutine_allocators {
+public:
+    static void* operator new(size_t size) {
+        memory::scoped_critical_alloc_section _;
+        return ::malloc(size);
+    }
+    static void operator delete(void* ptr) noexcept {
+        ::free(ptr);
```

Comment:
why in one branch ::free and memory::free in other?

Reply:
In this branch, we don't have the size, so we can't call sized deallocation.